File: career/frontend/types/advisor.ts
// Advisor Types - matches advisor-service (port 8014) schemas

export type InterventionType =
  | 'meeting'    // 상담
  | 'email'      // 이메일
  | 'referral'   // 의뢰/연계
  | 'resource'   // 자료 제공
  | 'followup';  // 후속 조치

export type InterventionStatus =
  | 'planned'
  | 'in_progress'
  | 'completed'
  | 'cancelled';

// Advisor Assignment
export interface AdvisorAssignmentResponse {
  assignment_id: string;
  advisor_id: string;
  advisor_name?: string;
  student_id: string;
  student_name?: string;
  assigned_at: string;
  is_primary: boolean;
  notes?: string;
}

// Intervention
export interface InterventionCreate {
  advisor_id: string;
  student_id: string;
  intervention_type: InterventionType;
  title: string;
  description?: string;
  scheduled_at?: string;
  priority?: number;  // 1-5
}

export interface InterventionUpdate {
  status?: InterventionStatus;
  outcome?: string;
  completed_at?: string;
  follow_up_needed?: boolean;
  next_action?: string;
}

export interface InterventionResponse {
  intervention_id: string;
  advisor_id: string;
  student_id: string;
  student_name?: string;
  intervention_type: InterventionType;
  title: string;
  description?: string;
  status: InterventionStatus;
  scheduled_at?: string;
  completed_at?: string;
  outcome?: string;
  follow_up_needed: boolean;
  next_action?: string;
  priority: number;
  created_at: string;
}

// Cohort Snapshot
export interface CohortSnapshotResponse {
  snapshot_id: string;
  advisor_id: string;
  snapshot_date: string;
  total_students: number;
  at_risk_count: number;
  on_track_count: number;
  excelling_count: number;
  avg_gpa: number;
  avg_progress: number;
  key_metrics: Record<string, unknown>;
  created_at: string;
}

// Advisor Note
export interface AdvisorNoteCreate {
  advisor_id: string;
  student_id: string;
  note_type?: string;
  content: string;
  is_private?: boolean;
  tags?: string[];
}

export interface AdvisorNoteResponse {
  note_id: string;
  advisor_id: string;
  student_id: string;
  student_name?: string;
  note_type: string;
  content: string;
  is_private: boolean;
  tags?: string[];
  created_at: string;
  updated_at?: string;
}

// Student Summary
export interface StudentSummary {
  student_id: string;
  student_name: string;
  major?: string;
  grade: number;
  gpa: number;
  risk_level: 'low' | 'medium' | 'high' | 'critical';
  active_alerts: number;
  last_interaction?: string;
  progress_percentage: number;
  needs_attention: boolean;
}

// Cohort Analytics
export interface CohortAnalytics {
  total_students: number;
  by_risk_level: Record<string, number>;
  by_grade: Record<number, number>;
  avg_gpa: number;
  avg_progress: number;
  interventions_this_month: number;
  pending_followups: number;
}

// Cohort Dashboard
export interface CohortDashboard {
  advisor_id: string;
  advisor_name?: string;
  analytics: CohortAnalytics;
  students_needing_attention: StudentSummary[];
  recent_interventions: InterventionResponse[];
  upcoming_interventions: InterventionResponse[];
  recent_notes: AdvisorNoteResponse[];
  generated_at: string;
}

// Meeting Types
export type MeetingStatus = 'scheduled' | 'in_progress' | 'completed' | 'cancelled';

export interface MeetingCreate {
  advisor_id: string;
  student_id: string;
  title: string;
  description?: string;
  scheduled_at: string;
  duration_minutes?: number;
  location?: string;
  meeting_type?: 'in_person' | 'video' | 'phone';
}

export interface MeetingUpdate {
  title?: string;
  description?: string;
  scheduled_at?: string;
  duration_minutes?: number;
  location?: string;
  meeting_type?: 'in_person' | 'video' | 'phone';
  status?: MeetingStatus;
  notes?: string;
}

export interface MeetingResponse {
  meeting_id: string;
  advisor_id: string;
  student_id: string;
  student_name?: string;
  title: string;
  description?: string;
  scheduled_at: string;
  duration_minutes: number;
  location?: string;
  meeting_type: 'in_person' | 'video' | 'phone';
  status: MeetingStatus;
  notes?: string;
  completed_at?: string;
  created_at: string;
}

// Type aliases for backward compatibility
export type AdvisorDashboardResponse = CohortDashboard;
export type StudentSummaryResponse = StudentSummary;
export type NoteCreate = AdvisorNoteCreate;
export type NoteResponse = AdvisorNoteResponse;

// Helper
export const INTERVENTION_TYPE_LABELS: Record<InterventionType, string> = {
  meeting: '상담',
  email: '이메일',
  referral: '의뢰/연계',
  resource: '자료 제공',
  followup: '후속 조치',
};

export const INTERVENTION_STATUS_LABELS: Record<InterventionStatus, string> = {
  planned: '예정',
  in_progress: '진행 중',
  completed: '완료',
  cancelled: '취소',
};

export const INTERVENTION_TYPE_ICONS: Record<InterventionType, string> = {
  meeting: '👥',
  email: '📧',
  referral: '🔗',
  resource: '📁',
  followup: '🔄',
};

File: career/services/advisor-service/app/schemas/advisor.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import date, datetime
from uuid import UUID
from enum import Enum


class InterventionType(str, Enum):
    MEETING = "meeting"           # 상담
    EMAIL = "email"               # 이메일
    REFERRAL = "referral"         # 의뢰/연계
    RESOURCE = "resource"         # 자료 제공
    FOLLOWUP = "followup"         # 후속 조치


class InterventionStatus(str, Enum):
    PLANNED = "planned"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"


class AdvisorAssignmentResponse(BaseModel):
    assignment_id: UUID
    advisor_id: UUID
    advisor_name: Optional[str] = None
    student_id: str
    student_name: Optional[str] = None
    assigned_at: datetime
    is_primary: bool = True
    notes: Optional[str] = None

    class Config:
        from_attributes = True


class InterventionCreate(BaseModel):
    advisor_id: str
    student_id: str
    intervention_type: InterventionType
    title: str
    description: Optional[str] = None
    scheduled_at: Optional[datetime] = None
    priority: int = Field(default=2, ge=1, le=5)


class InterventionUpdate(BaseModel):
    status: Optional[InterventionStatus] = None
    outcome: Optional[str] = None
    completed_at: Optional[datetime] = None
    follow_up_needed: Optional[bool] = None
    next_action: Optional[str] = None


class InterventionResponse(BaseModel):
    intervention_id: UUID
    advisor_id: str
    student_id: str
    student_name: Optional[str] = None
    intervention_type: InterventionType
    title: str
    description: Optional[str] = None
    status: InterventionStatus
    scheduled_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    outcome: Optional[str] = None
    follow_up_needed: bool = False
    next_action: Optional[str] = None
    priority: int
    created_at: datetime

    class Config:
        from_attributes = True


class CohortSnapshotResponse(BaseModel):
    snapshot_id: UUID
    advisor_id: str
    snapshot_date: date
    total_students: int
    at_risk_count: int
    on_track_count: int
    excelling_count: int
    avg_gpa: float
    avg_progress: float
    key_metrics: Dict[str, Any] = Field(default_factory=dict)
    created_at: datetime

    class Config:
        from_attributes = True


class AdvisorNoteCreate(BaseModel):
    advisor_id: str
    student_id: str
    note_type: str = "general"
    content: str
    is_private: bool = False
    tags: Optional[List[str]] = None


class AdvisorNoteResponse(BaseModel):
    note_id: UUID
    advisor_id: str
    student_id: str
    student_name: Optional[str] = None
    note_type: str
    content: str
    is_private: bool
    tags: Optional[List[str]] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class StudentSummary(BaseModel):
    student_id: str
    student_name: str
    major: Optional[str] = None
    grade: int
    gpa: float
    risk_level: str  # low, medium, high, critical
    active_alerts: int = 0
    last_interaction: Optional[datetime] = None
    progress_percentage: float = 0
    needs_attention: bool = False


class CohortAnalytics(BaseModel):
    total_students: int
    by_risk_level: Dict[str, int]
    by_grade: Dict[int, int]
    avg_gpa: float
    avg_progress: float
    interventions_this_month: int
    pending_followups: int


class CohortDashboard(BaseModel):
    advisor_id: str
    advisor_name: Optional[str] = None
    analytics: CohortAnalytics
    students_needing_attention: List[StudentSummary]
    recent_interventions: List[InterventionResponse]
    upcoming_interventions: List[InterventionResponse]
    recent_notes: List[AdvisorNoteResponse]
    generated_at: datetime


# Meeting Types
class MeetingStatus(str, Enum):
    SCHEDULED = "scheduled"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"


class MeetingType(str, Enum):
    IN_PERSON = "in_person"
    VIDEO = "video"
    PHONE = "phone"


class MeetingCreate(BaseModel):
    advisor_id: str
    student_id: str
    title: str
    description: Optional[str] = None
    scheduled_at: datetime
    duration_minutes: int = Field(default=30, ge=15, le=180)
    location: Optional[str] = None
    meeting_type: MeetingType = MeetingType.IN_PERSON


class MeetingResponse(BaseModel):
    meeting_id: UUID
    advisor_id: str
    student_id: str
    student_name: Optional[str] = None
    title: str
    description: Optional[str] = None
    scheduled_at: datetime
    duration_minutes: int
    location: Optional[str] = None
    meeting_type: MeetingType
    status: MeetingStatus
    notes: Optional[str] = None
    completed_at: Optional[datetime] = None
    created_at: datetime

    class Config:
        from_attributes = True
